pkg/querier/queryrange/downstreamer.go

```diff
@@ -131,11 +131,14 @@ func (in instance) For(
 
 	results := make([]logqlmodel.Result, len(queries))
 	for i := 0; i < len(queries); i++ {
-		resp := <-ch
-		if resp.err != nil {
-			return nil, resp.err
+		select {
+		case <-ctx.Done():
```

Comment:
Q: Should we put some info or error log (recording `ctx.Error()`) when we know we stop because of `ctx.Done()`? 

Wouldn't it help to understand why we got partial results in those cases?

Reply:
good catch!

Reply:
Since he now returns the error it's the caller that can log and do whatever he wants with the error.